### PR TITLE
Use mem::take instead of mem::replace with default

### PR DESCRIPTION
**Pull request by czipperz**

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
cc https://github.com/rust-lang/rust/issues/61129

r? @SimonSapin

**Review by dtolnay — APPROVED** [review]
Thanks!

**Comment by Centril** [comment]
@dtolnay Ha! I should have been faster! =)

File: src/tools/compiletest/src/runtest.rs [hunk 1]
```diff
@@ -3608,7 +3608,7 @@ fn nocomment_mir_line(line: &str) -> &str {
 
 fn read2_abbreviated(mut child: Child) -> io::Result<Output> {
     use crate::read2::read2;
-    use std::mem::replace;
+    use std::mem::take;
```
Comment (czipperz) [inline src/tools/compiletest/src/runtest.rs]: @dtolnay Any thoughts why this doesn't compile?  I have the `mem_take` feature enabled in this crate.  Wack

Comment (czipperz) [reply src/tools/compiletest/src/runtest.rs]: I think `compiletest` seems to be built with a stable compiler that doesn't have `take` yet.

Comment (Centril) [reply src/tools/compiletest/src/runtest.rs]: cc @pietroalbini

Comment (czipperz) [reply src/tools/compiletest/src/runtest.rs]: This change will probably need to be reverted until 1.36.0 (which should be released next week I believe) as `take` was added the first day of 1.35.0.

Comment (Mark-Simulacrum) [reply src/tools/compiletest/src/runtest.rs]: We intentionally build compiletest and several other tools with the bootstrap compiler (current beta).

Comment (Centril) [reply src/tools/compiletest/src/runtest.rs]: @czipperz Just skip the changes to compiletest for now and make them once we have a new beta compiler. To not forget, leave a note for yourself on the tracking issue. :)

Comment (czipperz) [reply src/tools/compiletest/src/runtest.rs]: KK

**Comment by Centril** [comment]
@bors r-

Seems you have some build failures in rustdoc.

**Comment by czipperz** [comment]
(rebased to fix merge errors)

**Pull request merged**